src/main/java/org/generationcp/middleware/util/uid/UIDGenerator.java

```diff
@@ -0,0 +1,70 @@
+package org.generationcp.middleware.util.uid;
+
+import com.google.common.base.Preconditions;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.apache.poi.ss.formula.functions.T;
+import org.generationcp.middleware.pojos.workbench.CropType;
+import org.springframework.util.CollectionUtils;
+
+import java.util.List;
+import java.util.UUID;
+
+public final class UIDGenerator {
+
+	public static enum UID_ROOT {
+		LOT("L"),
```

Comment:
dont we have same generation strategy for samples? 
was this change tested as a regression?

Reply:
As we discussed, I tested the generation for observation units, lots and germplasm, and logged the task to implement for samples in https://ibplatform.atlassian.net/browse/IBP-4167